mmdet/apis/ote/extension/datasets/mmdataset.py

```diff
@@ -51,16 +52,18 @@ def get_annotation_mmdet_format(
     for annotation in dataset_item.get_annotations(labels=labels, include_empty=False):
 
         box = ShapeFactory.shape_as_rectangle(annotation.shape)
-        polygon = ShapeFactory.shape_as_polygon(annotation.shape)
+
         class_indices = [
             label_idx[label.id]
             for label in annotation.get_labels(include_empty=False)
-            if label.domain == Domain.DETECTION
+            if label.domain == domain
         ]
 
         n = len(class_indices)
         gt_bboxes.extend([[box.x1 * width, box.y1 * height, box.x2 * width, box.y2 * height] for _ in range(n)])
-        gt_polygons.append([np.array([p for point in polygon.points for p in [point.x * width, point.y * height]])])
+        if domain != Domain.DETECTION:
+            polygon = ShapeFactory.shape_as_polygon(annotation.shape)
+            gt_polygons.extend([[np.array([p for point in polygon.points for p in [point.x * width, point.y * height]])] for _ in range(n)])
```

Comment:
It's a bit hard to read this. It might be better to separately convert polygon to a flat array of points.